### PR TITLE
feat(sindi_v2): add filter callback limit

### PR DESCRIPTION
**Pull request by Roxanne0321**
## Change Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement/Refactor
- [ ] Documentation
- [ ] CI/Build/Infra

## Linked Issue

Follow-up to #2680, extending the same callback-budget behavior to SINDI_V2.

## What Changed

- add the `sindi_v2.filter_callback_limit` search parameter with `0` as unlimited
- enforce the request-local callback budget in KNN and range search across mutable/immutable, distance-array, term-list, and empty-effective-query paths
- stop candidate and window scanning after processing the final allowed callback
- add parameter validation, behavioral coverage, and synchronized English/Chinese documentation

## Test Evidence

- [ ] `make fmt`
- [ ] `make lint`
- [ ] `make test`
- [ ] `make cov`, run tests, and collect coverage
- [x] Other (describe below)

Test details:

```text
clang-format 15.0.7 --dry-run --Werror <changed C++ files>
CMAKE_PREFIX_PATH=/root/vsag-disksindi/build/openblas/install make test-module MODULE=algorithm UT_FILTER="[SINDIV2]" COMPILE_JOBS=4
All tests passed (1750 assertions in 26 test cases)

build/tests/unittests_algorithm -d yes "[SINDIV2Parameter]" --allow-running-no-tests
All tests passed (76 assertions in 15 test cases)

git diff --check
```

## Compatibility Impact

- API/ABI compatibility: additive search parameter only; no index-format changes
- Behavior changes: only filtered searches with a positive limit terminate early and may return partial results; omitted or zero preserves existing behavior

## Performance and Concurrency Impact

- Performance impact: positive limits bound expensive user callbacks and stop later candidate/window scans; the default unlimited path avoids limiter allocation
- Concurrency/thread-safety impact: none; the remaining budget is request-local

## Documentation Impact

- [ ] No docs update needed
- [x] Updated docs:
  - [ ] `README.md`
  - [ ] `DEVELOPMENT.md`
  - [ ] `CONTRIBUTING.md`
  - [x] Other: SINDI_V2 index pages and index-parameter references in English and Chinese

## Risk and Rollback

- Risk level: low
- Rollback plan: revert commit `5b6b9e9f`

## Checklist

- [x] I have linked the relevant issue (required for `kind/bug` and `kind/feature`; see "Linked Issue" above)
- [x] I have added/updated tests for new behavior or bug fixes
- [x] I have considered API compatibility impact
- [x] I have updated docs if behavior/workflow changed
- [x] My commit messages follow project conventions (Conventional Commits, optional `[skip ci]` prefix)

### CODE REVIEW

File: src/algorithm/sindi_v2/sindi_v2.cpp [hunk 1]
```diff
@@ -54,6 +54,51 @@ constexpr const char* SINDI_V2_TERM_LAYOUT_KIND_KEY = "sindi_v2_term_layout_kind
 constexpr const char* SINDI_V2_TERM_LAYOUT_KIND = "term";
 constexpr uint64_t TERM_ID_MAPPER_ENTRY_MEMORY_BYTES = 54;
 
+class FilterCallbackLimiter : public Filter {
+public:
```
Comment (LHT129) [inline src/algorithm/sindi_v2/sindi_v2.cpp]: [note] The `FilterCallbackLimiter` class and `create_filter_callback_limiter` function are duplicated from `src/algorithm/sindi/sindi.cpp` (added in PR #2680). Consider extracting them to a shared location under `src/impl/filter/` to avoid maintaining two identical copies.

The two implementations are character-by-character identical. If a bug is found or the `Filter` interface gains a new virtual method, both copies would need to be updated independently.

Comment (Roxanne0321) [reply src/algorithm/sindi_v2/sindi_v2.cpp]: Addressed in 4179a379f64ae8d81abc1fb0168538d464ba0e26. I moved FilterCallbackLimiter and create_filter_callback_limiter into src/impl/filter/filter_callback_limiter.h, and both SINDI and SINDI_V2 now use that shared implementation. Focused validation passed: the algorithm test target built successfully; SINDI and SINDI_V2 filter callback limit tests, SINDI_V2 parameter validation, and the SINDI_V2 timeout regression all passed; clang-format 15 dry-run also passed.

**Review by LHT129 — COMMENTED** [review]
Reviewed commit 4179a379. The refactoring of `FilterCallbackLimiter` into `src/impl/filter/filter_callback_limiter.h` (addressing the previous review comment) is clean — both SINDI and SINDI_V2 now share the same implementation. The SINDI_V2 `filter_callback_limit` integration across KnnSearch, RangeSearch, and all three search paths (empty-effective-query, term-lists-heap-insert, dists-based) is correct and consistent. Parameter parsing, validation, round-trip serialization, and edge cases (zero, max uint64, negative, float) are well covered. Test coverage includes immutable/mutable, KNN/range, pruned/unpruned, remapped/direct term IDs, and the empty-query path. No blocking issues found.

**Review by wxyucs — APPROVED** [review]
lgtm
